README.md

```diff
@@ -1,2 +1,555 @@
+  --times TIMES  how many times to print
+```
+
+## Install
+
+```
+$ pip install {pipname}
+```
+
+## Getting started
+
+_Note: the examples are not [PEP 8](https://www.python.org/dev/peps/pep-0008/#blank-lines) compatible: one blank line is used instead of two to separate top-level definitions._
+
+In general, writing a CLI app is very similar to writing a regular function.
+{Name} is based on this metaphor, allowing to describe the whole interface of the app using only a function signature.
+Consider the following example:
+```py
+# print.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(phrase: Arg):
+    print(phrase)
+
+if __name__ == '__main__':
+    cli()
+```
+
+The script can be executed as a command-line app:
+```
+$ python3 print.py "Hello, world!"
+Hello, world!
+```
+
+The main idea is very simple: you use the `cli` decorator to wrap a function that acts as an entry point of the application (`entry`), and then use the `cli()` to make things running.
+`Arg` is used to annotate positional arguments of the CLI.
+
+In the following section, we'll discuss more deeply how to implement different types of parameters in {Name}.
+
+### Parameters
+
+There are three different annotations that represent positional arguments, flags, and options.
+
+#### `Arg`
+
+`Arg` is an annotation for positional arguments.
+In CLI, positional arguments work in the same way as in programming languages.
+
+Consider this function:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(x: Arg, y: Arg):
+    print(x, y)
+
+if __name__ == '__main__':
+    cli()
+```
+
+It can be called from a shell as follows:
+```
+$ python test.py 1 2
+1 2
+```
+
+But this call is invalid:
+```
+$ python test.py 1
+```
+
+It's possible though to provide a default value to the argument:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(x: Arg, y: Arg = 2):
+    print(x, y)
+
+if __name__ == '__main__':
+    cli()
+```
+
+To support the following:
+```
+$ python test.py 1
+1 2
+```
+
+Note: it's not possible to define something like this:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(x: Arg = 1, y: Arg):
+    print(x, y)
+
+if __name__ == '__main__':
+    cli()
+```
+
+The function signature is not supported even in Python.
+
+#### `Flag`
+
+`Flag` is an annotation for flags.
+In CLI, flags are boolean arguments that represent a turned off or turned on behavior.
+Unlike positional ones, they could be specified in a command line only with a special syntax.
+
+For example, the following function signature:
+```py
+# test.py
+
+from {pyname} import cli, Flag
+
+@cli
+def entry(some: Flag):
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+Could be called from the command line as follows:
+```
+$ python test.py --some
+True
+```
+
+But it's also possible to invoke the CLI without the flag:
+```
+$ python test.py
+False
+```
+
+Flags also support default values, but that doesn't make much sense.
```

Comment:
Do you think we need this example if it doesn't make sense?

Reply:
Removed.